go/common/errors/errors.go

```diff
@@ -0,0 +1,83 @@
+// Package errors implements errors that can be easily sent accross the
+// wire and reconstructed at the other end.
+package errors
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+)
+
+// UnknownModule is the module name used when the module is unknown.
+const UnknownModule = "unknown"
+
+var errUnknownError = New(UnknownModule, 1, "unknown error")
+
+// Re-exports so this package can be used as a replacement for errors.
+var (
+	As     = errors.As
+	Is     = errors.Is
+	Unwrap = errors.Unwrap
+)
+
+var registeredErrors sync.Map
+
+type codedError struct {
+	module string
+	code   uint32
+	msg    string
+}
+
+func (e *codedError) Error() string {
+	return e.msg
+}
+
+// New creates a new error.
+//
+// Module and code must be unique. If they are not, this method will
```

Comment:
Maybe something like `Module and code pair must be unique` might be more correct. (something that states that the pair of module & code needs to be unique, not each on it's own)